crates/mfem-sys/build.rs

```diff
@@ -17,9 +14,8 @@ fn main() {
         println!("cargo:rustc-link-lib={lib_type}={lib}");
     }
 
-    if is_windows {
-        println!("cargo:rustc-link-lib=dylib=user32");
-    }
+    #[cfg(target_os = "windows")]
+    println!("cargo:rustc-link-lib=dylib=user32");
```

Comment:
There is a subtle difference here because of the possibility of cross-compilation.

The info available to the `#[cfg(..)]` macro in `build.rs` is always about the build platform.
Whereas the info coming from the env vars is guaranteed to be about the target platform.
When the build and the target are the same, this is not an issue, but I want to err on the safe side.

Would anybody want to cross-compile `mfem-rs`? I don't know...
But I have a strong preference to be consistent with the larger Rust community.

See also this issue: https://github.com/rust-lang/rust/issues/43163

Reply:
Makes sense now that you mention it.  I've removed the commit.